test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/Hubs.cs

```diff
@@ -29,6 +29,26 @@ public class TestHub : Hub
             await Clients.Client(Context.ConnectionId).InvokeAsync("Echo", message);
         }
 
+        public int GetLocalPort()
```

Comment:
A single method call with all of the data would be sufficient.

PS: This hub is becoming too large and we need to write more unit tests.